fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/BasicCacheEntry.java

```diff
@@ -17,6 +17,7 @@
  */
 package org.fcrepo.kernel.modeshape.utils;
 
+import org.apache.http.impl.auth.UnsupportedDigestAlgorithmException;
```

Comment:
I believe this should be `org.fcrepo.kernel.api.exception.UnsupportedAlgorithmException`.

When I do:
`curl -H "Want-Digest: sha256" -I http://localhost:8080/rest/img.jpg`

I get a 500 error, caught by the wildcard exception mapper.  The exception is:
<pre>
DEBUG 16:53:31.225 (WildcardExceptionMapper) WildcardExceptionMapper intercepted exception:{}

org.apache.http.impl.auth.UnsupportedDigestAlgorithmException: Unsupported digest algorithm: sha256
        at org.fcrepo.kernel.modeshape.utils.BasicCacheEntry.checkFixity(BasicCacheEntry.java:111)
        at org.fcrepo.kernel.modeshape.FedoraBinaryImpl.checkFixity(FedoraBinaryImpl.java:392)
        at org.fcrepo.http.api.FedoraLdp.handleWantDigestHeader(FedoraLdp.java:802)
        at org.fcrepo.http.api.FedoraLdp.head(FedoraLdp.java:217)
</pre>

Reply:
Good catch! Thanks. I've added a commit to change the exception to UnsupportedAlgorithmException with a test. See https://github.com/fcrepo4/fcrepo4/pull/1231/commits/c306e2f9d6cfa2f7241becbe7f29f1cd1ed6f824.